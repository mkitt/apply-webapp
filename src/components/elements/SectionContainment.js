// @flow
import glamorous from 'glamorous'

const rules = {
  position: 'relative',
  maxWidth: 960,
  marginRight: 'auto',
  marginLeft: 'auto',
}

export default glamorous.section(
  rules,
  ({ children, ...props }) => ({ ...props }),
)
