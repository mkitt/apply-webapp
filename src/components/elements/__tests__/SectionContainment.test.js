import React from 'react'
import renderer from 'react-test-renderer'
import SectionContainment from '../SectionContainment'

describe('<SectionContainment />', () => {
  it('shows the correct snapshot tree for the <SectionContainment> element', () => {
    const component = <SectionContainment />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <SectionContainment> element with options', () => {
    const component = (
      <SectionContainment my="auto" px={24} py={24} />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
