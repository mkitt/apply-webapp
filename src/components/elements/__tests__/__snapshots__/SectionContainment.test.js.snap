// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainment /> shows the correct snapshot tree for the <SectionContainment> element 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 16px;
  padding-left: 16px;
}

<section
  className="css-0"
/>
`;

exports[`<SectionContainment /> shows the correct snapshot tree for the <SectionContainment> element with options 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 24px;
  padding-left: 24px;
  margin-top: auto;
  margin-bottom: auto;
  padding-top: 24px;
  padding-bottom: 24px;
}

<section
  className="css-0"
/>
`;
